docs/docs/spark-functions.md

```diff
@@ -0,0 +1,262 @@
+
+### `system.iceberg_version()`
+
+Returns the Iceberg library version at runtime.
+
+#### Description
+Returns the current version of Iceberg used in the Spark runtime. This is useful for debugging, validation, or auditing purposes where it's necessary to track the version of Iceberg being used.
+
+#### Usage
+
+```sql
+SELECT system.iceberg_version();
+```
+
+#### Returns
+A `STRING` value representing the Iceberg version, e.g., `"1.5.0"`.
+
+#### Example
+
+```sql
+SELECT system.iceberg_version();
+-- Result: 1.5.0
+```
+
+---
+
+### `system.bucket(value, num_buckets)`
+
+Returns the bucket number for a given value, using the same hash logic as Iceberg's `bucket` partition transform.
+
+---
+
+#### Description
+Computes a deterministic bucket number for the given value, compatible with Iceberg table partitioning. This is useful when validating partition logic or computing bucket values externally (e.g., in ETL pipelines).
+
+#### Arguments
+
+- `value`: A primitive type (e.g., `INT`, `STRING`, `BIGINT`).
+- `num_buckets`: Positive integer indicating the number of buckets.
+
+#### Usage
+
+```sql
+SELECT system.bucket(column_value, num_buckets);
+```
+
+#### Parameters
+
+* `column_value`: Any supported primitive type (e.g., `INT`, `STRING`) to be hashed.
+* `num_buckets`: The total number of buckets (positive integer).
+
+#### Returns
+An `INT` between `0` and `num_buckets - 1`.
+
+#### Example
+
+```sql
+SELECT system.bucket('user_id_123', 8);
+-- Result: 3
+```
+
+### `system.years(value)`
+
+Returns the number of years since the Unix epoch (1970) for a given date or timestamp.
+
+#### Description
+
+Calculates `year(value) - 1970`. This is consistent with the Iceberg `years` partition transform, and can be used to compute partition keys manually.
+
+#### Arguments
+
+- `value`: A `DATE`, `TIMESTAMP`, or `TIMESTAMP_NTZ`. Other types will result in an error.
+
+#### Returns
+
+An `INT` representing the number of years from 1970. If the input is `NULL`, the result is also `NULL`.
+
+#### Examples
+
+```sql
+SELECT system.years(DATE '2017-12-01');
+-- Result: 47
+
+SELECT system.years(TIMESTAMP '1969-12-31 23:59:59');
+-- Result: -1
+```
+
+### `system.months(value)`
+
+Returns the number of months since the Unix epoch (1970-01-01) for a given date or timestamp.
+
+#### Description
+
+Calculates `12 * (year(value) - 1970) + month(value) - 1`. This is consistent with the Iceberg `months` partition transform and can be used to compute partition keys manually.
+
+#### Arguments
+
+- `value`: A `DATE`, `TIMESTAMP`, or `TIMESTAMP_NTZ`. Other types will result in an error.
+
+#### Returns
+
+An `INT` representing the number of months from January 1970. If the input is `NULL`, the result is also `NULL`.
+
+#### Examples
+
+```sql
+SELECT system.months(DATE '2017-12-01');
+-- Result: 575
+
+SELECT system.months(TIMESTAMP '1969-12-31 23:59:59');
```

Comment:
maybe also add an example with 1970, similarly to https://github.com/apache/iceberg/blob/6c98a9e470d44f77e1939a1a333bf197a57949a6/spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkMonthsFunction.java#L45